common/metering.go

```diff
@@ -52,36 +52,6 @@ type ComputationGauge interface {
 	MeterComputation(usage ComputationUsage) error
 }
 
-// Gauge combines a memory and computation gauge.
-// Metering-sites can use this combined interface
-// if they need to do both memory and computation metering.
-type Gauge interface {
```

Comment:
Removed the combined gauge, because it adds another layer of indirection, and thus overhead: Both memory gauge and computation gauge are optional, so the implementations of the methods `MeterMemory` and `ComputationGauge` would actually have to check for `nil` each time they are called.